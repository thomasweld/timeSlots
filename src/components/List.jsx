import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import * as listActions from '../ducks/listDuck';
import TimeSlotForm from './TimeSlotForm';

class List extends React.Component {
  render() {
    let showList, showModal;
    if (this.props.timeSlots) {
      showList = this.props.timeSlots.map((timeSlot, index) => (
        <li
          key={index}
          onClick={() => this.props.onClickTimeSlot(timeSlot)}
          style={{
            backgroundColor: timeSlot.available ? 'lightgreen' : 'red'
          }}
        >
          {timeSlot.hour}
          {!timeSlot.available &&
            ` - Booked by ${timeSlot.name} - ${timeSlot.phoneNumber}`}
        </li>
      ));
    }

    if (this.props.showModal && this.props.activeTimeSlot) {
      showModal = (
        <div>
          <h4>Sign Up for {this.props.activeTimeSlot.hour}</h4>
          <TimeSlotForm
            onSubmit={this.props.handleSubmit}
            handleCancel={this.props.handleCancel}
            initialValues={{
              name: `${this.props.activeTimeSlot.name}`,
              phoneNumber: `${this.props.activeTimeSlot.phoneNumber}`,
              hour: `${this.props.activeTimeSlot.hour}`
            }}
          />
        </div>
      );
    }

    return (
      <div>
        <h1>Time Slots</h1>
        <div>{showModal}</div>
        <ul>{!showModal && showList}</ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  timeSlots: state.List.timeSlots,
  showModal: state.List.showModal,
  activeTimeSlot: state.List.activeTimeSlot
});

const mapDispatchToProps = dispatch => ({
  onClickTimeSlot: listActions.launchPopUp,
  handleSubmit: listActions.handleSubmit,
  handleCancel: listActions.handleCancel
});

List.propTypes = {
  timeSlots: PropTypes.array.isRequired,
  onClickTimeSlot: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps())(List);
