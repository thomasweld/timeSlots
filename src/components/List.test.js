import React from 'react';
import renderer from 'react-test-renderer';
import { List } from './List';

const initialState = {
  activeTimeSlot: {
    hour: '10am',
    available: true,
    name: 'John',
    phoneNumber: '1234566543'
  },
  timeSlots: [
    {
      hour: '9am',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '10am',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '11am',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '12pm',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '1pm',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '2pm',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '3pm',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '4pm',
      available: true,
      name: '',
      phoneNumber: ''
    },
    {
      hour: '5pm',
      available: true,
      name: '',
      phoneNumber: ''
    }
  ]
};

// Snapshot test example
describe('List', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <List
        timeSlots={initialState.timeSlots}
        onClickTimeSlot={() => {}}
        showModal={false}
        activeTimeSlot={initialState.activeTimeSlot}
        handleSubmit={() => {}}
        handleCancel={() => {}}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
