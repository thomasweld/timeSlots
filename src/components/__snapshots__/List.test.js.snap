// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<div>
  <h1>
    Time Slots
  </h1>
  <div />
  <ul>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      9am
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      10am
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      11am
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      12pm
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      1pm
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      2pm
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      3pm
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      4pm
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
    >
      5pm
    </li>
  </ul>
</div>
`;
